mono/mini/mini-arm.c

```diff
@@ -432,6 +429,94 @@ emit_move_return_value (MonoCompile *cfg, MonoInst *ins, guint8 *code)
 	return code;
 }
 
+static guint8*
+mono_arm_emit_tls_get (MonoCompile *cfg, guint8* code, int dreg, int tls_offset)
+{
+#if (defined(HAVE_KW_THREAD) && defined(__linux__)) \
```

Comment:
One last thing: Could we avoid these long lists of defines by moving it to the top of the file, like

    #if (defined (HAVE_KW_THREAD) && defined (__linux__) || defined (TARGET_ANDROID) || defined (TARGET_IOS)
    #define USE_FAST_TLS
    #endif

and then use `#ifdef USE_FAST_TLS` in the rest of the code.